clang/lib/CConv/ConstraintsGraph.cpp

```diff
@@ -26,19 +26,26 @@ ConstraintsGraph::vertex_t ConstraintsGraph::addVertex(Atom *A) {
   if (ConstAtom *CA = clang::dyn_cast<ConstAtom>(A)) {
     AllConstAtoms.insert(CA);
   }
-
-  // If we haven't seen the Atom? Insert into the graph.
-  if (AtomToVDMap.find(A) == AtomToVDMap.end()) {
-    auto Vidx = add_vertex(A, CG);
-    AtomToVDMap[A] = Vidx;
-  }
-  return AtomToVDMap[A];
+  return BaseGraph::addVertex(A);
 }
 
 std::set<ConstAtom*> &ConstraintsGraph::getAllConstAtoms() {
   return AllConstAtoms;
 }
 
+std::set<std::pair<Atom*, Atom*>> ConstraintsGraph::getAllEdges() {
```

Comment:
This approach could be expensive: It creates a set of the edges, which is then immediately consumed and discarded to merge it into the GraphViz graph. What about, instead, doing something higher order? You pass in a function `f` to this function (call it `forAllEdges` or something like that) which will, for each edge `e` in `this`, call the function `f(e)`. The function can be the `addEdge` function for the merged graph you are creating.